pymlx/features_handler.py

```diff
@@ -111,16 +113,53 @@ def learn(self, df):
                     _map[value] = self._count
                     out_feature_names.append(in_feature_name + '=' + str(value))
                     self._count += 1
-            maps[i] = _map
+            # maps[i] = _map
+            maps[in_feature_name] = _map  # no more number index
 
     def apply(self, input_generator):
         preprocessor = self._preprocessor
         for i, value in enumerate(input_generator):
-            index = self._maps[i].get(preprocessor(value))
+            col, val = value
+            index = self._maps[col].get(preprocessor(val))
             if index is not None:
                 yield index, 1
 
 
+class IdEncodingHandler(FeaturesHandler):
+    """
+    Convert category to id
+    """
+    def __init__(self, in_feature_names, preprocessor=None):
+        super(self.__class__, self).__init__(in_feature_names)
+        maps = {}
+        self.out_feature_names = [name + '_ide' for name in in_feature_names] # ide = id encoding
+        self._maps = maps
+        self._count = 0
+        self._preprocessor = lambda x: x if preprocessor is None else preprocessor
+
+    def learn(self, df):
+        maps = self._maps
+        preprocessor = self._preprocessor
+        for i, in_feature_name in enumerate(self.in_feature_names):
+            _map = {}
+            for value in df[in_feature_name]:
+                value = preprocessor(value)
+                if value not in _map:
+                    _map[value] = self._count
+                    # out_feature_names.append(in_feature_name + '=' + str(value))
```

Comment:
Do not leave commented code. We have version control so we can find old code easily, when needed.